flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionVertex.java

```diff
@@ -726,6 +728,56 @@ void sendPartitionInfos() {
 		}
 	}
 
+	/**
+	 * Check whether the InputDependencyConstraint is satisfied for this vertex.
+	 *
+	 * @return whether the input constraint is satisfied
+	 */
+	public boolean checkInputDependencyConstraints() {
```

Comment:
Thanks for the suggestion. I made it more concise, but a bit different from your sample code:
For PIPELINED input consumable check, only the partitions related with the input edges are checked. 

I'm not sure why you think the behavior is changed? 
I'd agree if you have concerns that this check could be heavy(O(N) complexity)) for large parallelism vertex. So I'm considering whether to add a short cut check for `InputDependencyConstraint.ANY` in `Execution.scheduleOrUpdateConsumers`, i.e. change `if (consumerVertex.checkInputDependencyConstraints()) ` to `if (getVertex().getExecutionGraph().getInputDependencyConstraint() == InputDependencyConstraint.ANY || consumerVertex.checkInputDependencyConstraints())`. It's not elegant but can be as efficient as current scheduling.

The input state is not used in EAGER scheduling. So the InputDependencyConstraint is not a matter in EAGER mode. It is used in scheduleOrUpdateConsumers, which is only invoked in LAZY_FROM_SOURCES scheduling.

Reply:
We can skip the input constraint check for `InputDependencyConstraint.ANY` because when invoking `scheduleOrUpdateConsumers`, at least one of the consumer vertex's inputs is consumable. This assumption may not be right if `checkInputDependencyConstraints` is invoked from other entries.
That's why I think it's better to add the shortcut in `scheduleOrUpdateConsumers`.